libraries/rush-lib/src/utilities/WebClient.ts

```diff
@@ -53,49 +57,196 @@ export enum WebClientProxy {
+      const parsedUrl: URL = typeof url === 'string' ? new URL(url) : url;
+      const requestFunction: typeof httpRequest | typeof httpsRequest =
+        parsedUrl.protocol === 'https:' ? httpsRequest : httpRequest;
+
+      requestFunction(url, options, (response: IncomingMessage) => {
+        const responseBuffers: (Buffer | Uint8Array)[] = [];
+        response.on('data', (chunk: string | Buffer | Uint8Array) => {
+          responseBuffers.push(Buffer.from(chunk));
+        });
+        response.on('end', () => {
+          // Handle retries by calling the method recursively with the redirect URL
+          const statusCode: number | undefined = response.statusCode;
+          if (statusCode === 301 || statusCode === 302) {
+            switch (redirect) {
+              case 'follow': {
+                const redirectUrl: string | string[] | undefined = response.headers.location;
+                if (redirectUrl) {
+                  makeRequestAsync(redirectUrl, options, true).then(resolve).catch(reject);
+                } else {
+                  reject(
+                    new Error(`Received status code ${response.statusCode} with no location header: ${url}`)
+                  );
+                }
+
+                break;
+              }
+              case 'error':
+                reject(new Error(`Received status code ${response.statusCode}: ${url}`));
+                return;
+            }
+          }
+
+          const responseData: Buffer = Buffer.concat(responseBuffers);
+          const status: number = response.statusCode || 0;
+          const statusText: string | undefined = response.statusMessage;
+          const headers: Record<string, string | string[] | undefined> = response.headers;
+
+          let bodyString: string | undefined;
+          let bodyJson: unknown | undefined;
+          let decodedBuffer: Buffer | undefined;
+          const result: IWebClientResponse = {
+            ok: status >= 200 && status < 300,
+            status,
+            statusText,
+            redirected,
+            headers,
+            getTextAsync: async () => {
+              if (bodyString === undefined) {
+                const buffer: Buffer = await result.getBufferAsync();
+                // eslint-disable-next-line require-atomic-updates
+                bodyString = buffer.toString();
+              }
+
+              return bodyString;
+            },
+            getJsonAsync: async <TJson>() => {
+              if (bodyJson === undefined) {
+                // eslint-disable-next-line require-atomic-updates
+                bodyJson = await result.getTextAsync();
+              }
+
+              return bodyJson as TJson;
+            },
+            getBufferAsync: async () => {
+              // Determine if the buffer is compressed and decode it if necessary
+              if (decodedBuffer === undefined) {
+                let encodings: string | string[] | undefined = headers['content-encoding'];
```

Comment:
Does this get parsed as a string array? Or do you need to manually split on the `,` and trim whitespace?